### PR TITLE
spglib@2.1+ requires cmake@3.24+

### LINKED ISSUE #42241: Installation issue: spglib cmake fails with cmake<3.24
### Steps to reproduce the issue

```console
$ spack spec -I spglib@2.2.0 ^cmake@3.23.4
Concretized
--------------------------------
 -   spglib@2.2.0%gcc@12.3.0+fortran~ipo+openmp build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-skylake
[+]      ^cmake@3.23.4%gcc@12.3.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-ubuntu22.04-skylake
[+]          ^curl@8.4.0%gcc@12.3.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-ubuntu22.04-skylake
[+]              ^nghttp2@1.57.0%gcc@12.3.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]              ^openssl@3.1.3%gcc@12.3.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu22.04-skylake
[+]                  ^ca-certificates-mozilla@2023-05-30%gcc@12.3.0 build_system=generic arch=linux-ubuntu22.04-skylake
[+]                  ^perl@5.38.0%gcc@12.3.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-ubuntu22.04-skylake
[+]                      ^berkeley-db@18.1.40%gcc@12.3.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu22.04-skylake
[+]                      ^bzip2@1.0.8%gcc@12.3.0~debug~pic+shared build_system=generic arch=linux-ubuntu22.04-skylake
[e]                          ^diffutils@3.8%gcc@12.3.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]                      ^gdbm@1.23%gcc@12.3.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]                          ^readline@8.2%gcc@12.3.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu22.04-skylake
[+]              ^pkgconf@1.9.5%gcc@12.3.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]          ^ncurses@6.4%gcc@12.3.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu22.04-skylake
[+]          ^zlib-ng@2.1.5%gcc@12.3.0+compat+opt build_system=autotools arch=linux-ubuntu22.04-skylake
[+]      ^gcc-runtime@12.3.0%gcc@12.3.0 build_system=generic arch=linux-ubuntu22.04-skylake
[+]      ^gmake@4.4.1%gcc@12.3.0~guile build_system=generic arch=linux-ubuntu22.04-skylake
```

### Error message

<details><summary>Error message</summary>

<pre>
==> Installing spglib-2.2.0-kjdh2zl3kgvxhpplhrxsq5imjateho42 [9/9]
==> No binary for spglib-2.2.0-kjdh2zl3kgvxhpplhrxsq5imjateho42 found: installing from source
==> Using cached archive: /home/lercole/src/spack-dev/var/spack/cache/_source-cache/archive/ac/ac929e20ec9d4621411e2cdec59b1442e02506c1e546005bbe2c7f781e9bd49a.tar.gz
==> No patches needed for spglib
==> spglib: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.22.1-tifrz6pqgt4ckxerucqldqx3wpnt4fcb/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/spglib-2.2.0-kjdh2zl3kgvxhpplhrxsq5imjateho42' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/spglib-2.2.0-kjdh2zl3kgvxhpplhrxsq5imjateho42/lib;/home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/spglib-2.2.0-kjdh2zl3kgvxhpplhrxsq5imjateho42/lib64;/home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gcc-runtime-12.3.0-pvfklhweoh6sjj3mrb2q5prmepkvwqiy/lib' '-DCMAKE_PREFIX_PATH:STRING=/home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.22.1-tifrz6pqgt4ckxerucqldqx3wpnt4fcb;/home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gmake-4.4.1-454qjr4n5agjmypsynkbdfwn5mslmmi2;/home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gcc-runtime-12.3.0-pvfklhweoh6sjj3mrb2q5prmepkvwqiy' '-DSPGLIB_USE_OMP:BOOL=ON' '-DSPGLIB_WITH_Fortran:BOOL=ON' '/tmp/lercole/spack-stage/spack-stage-spglib-2.2.0-kjdh2zl3kgvxhpplhrxsq5imjateho42/spack-src'

5 errors found in build log:
     19    -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
     20    -- Found Threads: TRUE
     21    [ 11%] Creating directories for 'gtest-populate'
     22    [ 22%] Performing download step (git clone) for 'gtest-populate'
     23    -- Avoiding repeated git clone, stamp file is up to date: '/tmp/lercole/spack-stage/spack-stage-spglib-2.2.0-kjdh2zl3kgvxhpplhrxsq5imjateho42/spack-build-kjdh2zl/_deps/gtest-subbuild/gtest-populate-prefix/src/gtest-populate-stamp/gtest-populate-gitinfo.txt'
     24    [ 33%] Performing update step for 'gtest-populate'
  >> 25    CMake Error at /tmp/lercole/spack-stage/spack-stage-spglib-2.2.0-kjdh2zl3kgvxhpplhrxsq5imjateho42/spack-build-kjdh2zl/_deps/gtest-subbuild/gtest-populate-prefix/tmp/gtest-populate-gitupdate.cmake:25 (message):
     26      Failed to get the hash for HEAD:
     27
     28
     29
  >> 30    gmake[2]: *** [CMakeFiles/gtest-populate.dir/build.make:136: gtest-populate-prefix/src/gtest-populate-stamp/gtest-populate-update] Error 1
  >> 31    gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/gtest-populate.dir/all] Error 2
  >> 32    gmake: *** [Makefile:91: all] Error 2
     33
  >> 34    CMake Error at /home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.22.1-tifrz6pqgt4ckxerucqldqx3wpnt4fcb/share/cmake-3.22/Modules/FetchContent.cmake:1087 (message):
     35      Build step for gtest failed: 2
     36    Call Stack (most recent call first):
     37      /home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.22.1-tifrz6pqgt4ckxerucqldqx3wpnt4fcb/share/cmake-3.22/Modules/FetchContent.cmake:1216:EVAL:2 (__FetchContent_directPopulate)
     38      /home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.22.1-tifrz6pqgt4ckxerucqldqx3wpnt4fcb/share/cmake-3.22/Modules/FetchContent.cmake:1216 (cmake_language)
     39      /home/lercole/src/spack-dev/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.22.1-tifrz6pqgt4ckxerucqldqx3wpnt4fcb/share/cmake-3.22/Modules/FetchContent.cmake:1259 (FetchContent_Populate)
     40      CMakeLists.txt:142 (FetchContent_MakeAvailable)

See build log for details:
  /tmp/lercole/spack-stage/spack-stage-spglib-2.2.0-kjdh2zl3kgvxhpplhrxsq5imjateho42/spack-build-out.txt
</pre></details>

### Information on your system

* **Spack:** 0.22.0.dev0 (3a1fa0c359dad8c545094b40eca3a575c59b0675)
* **Python:** 3.10.10
* **Platform:** linux-ubuntu22.04-skylake
* **Concretizer:** clingo

### Additional information

It seems to me that spglib 2.1.0+ needs cmake 3.24+. I will open a PR with this fix.
@RemiLacroix-IDRIS

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by lorisercole**
Fixes #42241

### CODE REVIEW

**Comment by RemiLacroix-IDRIS** [comment]
See https://github.com/spack/spack/issues/42241#issuecomment-1905851214, I'm not completely sure the problem is CMake related.

**Comment by haampie** [comment]
```
spack-stage-spglib-2.2.0-khl3qppjgs4g3tchgrcz2u3qhtfj63bo $ rg cmake_minimum 
spack-src/CMakeLists.txt
1:cmake_minimum_required(VERSION 3.15)

spack-src/example/c_api/CMakeLists.txt
1:cmake_minimum_required(VERSION 3.6)

spack-src/example/fortran_api/CMakeLists.txt
1:cmake_minimum_required(VERSION 3.6)

spack-src/test/package/FetchContent/fortran/CMakeLists.txt
1:cmake_minimum_required(VERSION 3.11)

spack-src/test/package/FetchContent/c/CMakeLists.txt
1:cmake_minimum_required(VERSION 3.11)

spack-src/test/package/pkg-config/fortran/CMakeLists.txt
1:cmake_minimum_required(VERSION 3.11)

spack-src/test/package/pkg-config/c/CMakeLists.txt
1:cmake_minimum_required(VERSION 3.11)

spack-src/test/package/find_package/c/CMakeLists.txt
1:cmake_minimum_required(VERSION 3.11)

spack-src/test/package/find_package/fortran/CMakeLists.txt
1:cmake_minimum_required(VERSION 3.11)

spack-build-khl3qpp/_deps/gtest-subbuild/CMakeLists.txt
4:cmake_minimum_required(VERSION 3.22.6)

spack-build-khl3qpp/_deps/gtest-subbuild/gtest-populate-prefix/tmp/gtest-populate-gitupdate.cmake
4:cmake_minimum_required(VERSION 3.5)
```
so `gtest-subbuild` is the offender.

You could also just disable tests `SPGLIB_WITH_TESTS=OFF`.

**Comment by RMeli** [comment]
> You could also just disable tests SPGLIB_WITH_TESTS=OFF.

It would be nice to add a `test` variant (defaulting to `False`, given the problem), if you are up for it.

Not sure what's the best way around this issue (@haampie any suggestion?), but with such a `test` variant you could then make the `cmake@3.24:` requirement conditional on that variant.

**Comment by lorisercole** [comment]
> > You could also just disable tests SPGLIB_WITH_TESTS=OFF.
> 
> It would be nice to add a `test` variant (defaulting to `False`, given the problem), if you are up for it.
> 
> Not sure what's the best way around this issue (@haampie any suggestion?), but with such a `test` variant you could then make the `cmake@3.24:` requirement conditional on that variant.

Yes, I also tried that as a workaround. We can make a `tests` variant with the additional constrain on cmake.
Should it be activated by default?

**Comment by RMeli** [comment]
> Should it be activated by default?

I'd say to make it `False` by default, given the issue. I had a look at a few other packages, and the `tests` variant is often off by default.

**Comment by RMeli** [comment]
But I don't have a strong preference either way TBH.

File: var/spack/repos/builtin/packages/spglib/package.py [hunk 1]
```diff
@@ -52,6 +52,9 @@ class Spglib(CMakePackage):
 
     variant("openmp", default=True, description="Build with OpenMP support", when="@1.16.2:")
     variant("fortran", default=True, description="Build Fortran interface", when="@1.16.4:")
+    variant("tests", default=False, description="Build with tests", when="@2:")
```
Comment (RMeli) [inline var/spack/repos/builtin/packages/spglib/package.py]:
```suggestion
    variant("tests", default=False, description="Build with tests", when="@2.1.0:")
```
If I'm not mistaken?

File: var/spack/repos/builtin/packages/spglib/package.py [hunk 2]
```diff
@@ -52,6 +52,9 @@ class Spglib(CMakePackage):
...
+
+    depends_on("cmake@3.24:", type="build", when="@2.1:+tests")
```
Comment (RMeli) [inline var/spack/repos/builtin/packages/spglib/package.py]:
```suggestion
    depends_on("cmake@3.24:", type="build", when="+tests")
```
See comment above.

**Review by RMeli — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/spglib/package.py (hunk 1 above)
Comment (lorisercole) [reply var/spack/repos/builtin/packages/spglib/package.py]: Oh you are right. I saw a `test` folder appearing from version 2.0, but the `SPGLIB_WITH_TESTS` CMake option is only implemented from v 2.1.

**Review by RMeli — DISMISSED** [review]
Thanks @lorisercole. I'll leave the final check and merge to @haampie.

**Review by haampie — DISMISSED** [review]

**Comment by haampie** [comment]
Wait, can you also add

```python
depends_on("cmake@3.15:", type="build", when="@2.1.0:")
```

for good measure?

**Comment by lorisercole** [comment]
> Wait, can you also add
> 
> ```python
> depends_on("cmake@3.15:", type="build", when="@2.1.0:")
> ```
> 
> for good measure?

Sure!

**Review by RMeli — APPROVED** [review]

**Pull request merged**